packages/gateway-client/src/worker/p2p-client/stream-factory.ts

```diff
@@ -139,4 +222,27 @@ export class StreamFactory {
     public async getStream(protocol = '/samizdapp-proxy') {
         return this.makeStream(protocol);
     }
+
+    public async getProxyStream() {
+        do {
+            const stream = this.availableStreams.pop();
+            if (stream?.isOpen) {
+                return stream;
+            }
+        } while (this.availableStreams.length > 0);
+
+        const wrapped = new WrappedStream(
+            await this.getStream('/samizdapp-proxy/2.0.0')
+        );
+        wrapped.open();
+        return wrapped;
+    }
+
+    public releaseProxyStream(stream: WrappedStream) {
```

Comment:
Ideally there could just be a `release()` method on the returned stream that is a no-op if the stream wasn't pooled to begin with. But if that is too complex or too much to implement, this is probably fine.